Accuse

NBlair52 opened the pull request:
Accuse Logic is in

I think the rebase went pretty smooth
On testing with the rebase move left immediatley threw an error on the server side, but it looks like accuse is still working.

TODO
-Setting Player to out doesn't really do anything (no checking on server side)

NBlair52 committed:
Accuse Logice Added to Server

NBlair52 committed:
Accuse Logic Added to Client

Minor Server Updates

NBlair52 committed:
Accuse Messages Added

NBlair52 committed:
Minor Changes to Server and Client

NBlair52 committed:
Added isEqual to SolutionHand

NBlair52 committed:
Accuse Working

Minor Fixes to all

Kit-Kindred commented on Source_Code/Common/Messages/ActionRequests/AccuseRequest.java:
```diff
@@ -0,0 +1,20 @@
+
+public class AccuseRequest extends ActionRequest
+{
+    public String PlayerName;
```
Added PlayerName to ActionRequest, so we don't have to have it here

NBlair52 replied:
As I was writing these 84 times I was like "inheritence should have taken care of this..." but was too lazy to check.

Kit-Kindred commented on Source_Code/Server/ClueLessServer.java:
```diff
@@ -467,6 +478,31 @@ else if ( ! ( possibleRefutations.getWeapons().isEmpty()) )
+        {
+            // If incorrect. tell everyone, set player "out"
+            accuseNotification.setCorrect(false);
+            // TODO - looks like setting this to false doesn't actually prevent them from making moves
```
ClueLessServer:nextMove will auto-skip disconnected and out players when assigning a turn. Need to do more testing to be sure it works 100%, but it seems to at least work at a rudimentary level. Need to add the turn tracking to the processAccuse, and I think everything'll work as expected (I can do this, just want to put the status here)

NBlair52 replied:
comment made before testing. Will take out now, I am sure we will notice if this does not work

Kit-Kindred commented on Source_Code/Server/ClueLessServer.java:
```diff
@@ -305,6 +310,12 @@ else if(actionRequest instanceof EndTurn || turnTracker.isTurnOver())  // isTurn
             }
         }
+
+        if (actionRequest instanceof AccuseRequest)
```
This needs to be moved into the "if(PlayerList.get( CurrentPlayerIndex ).PlayerName.equals(p.PlayerName))  // Only the player who's turn it is can do these" if statement. Don't want non-players to accidentally issue these

NBlair52 replied:
whoops

Kit-Kindred commented on Source_Code/Client/ClueLessClient.java:
```diff
@@ -362,6 +396,38 @@ else if( statUp instanceof SuggestionWrong)
+            }
+        }
+        // This is sent to a player to let them know they are out of the game.
+        // Do we have to do anything else here?
```
We should tell the out-player what the correct answer is (like how you get to see what's actually in the enevelope)

NBlair52 replied:
Love that. Adding now

Kit-Kindred commented on Source_Code/Client/ClueLessClient.java:
```diff
@@ -215,6 +215,39 @@ public static void main(String[] args)
                             clientApplication.csc.send(new EndTurn(clientApplication.UserPlayer.PlayerName));
                         }
+                // Putting this as 9 so that they dont hit it on accident because its final
+                case 9 ->
```
Won't let me add comments to non-changed lines, so I'll put it here: This Accuse: 9 should be added to the processTurnUpdate print outs (~line 493)

Kit-Kindred commented on Source_Code/Client/ClueLessClient.java:
```diff
@@ -362,6 +396,38 @@ else if( statUp instanceof SuggestionWrong)
         }
+
+        // Notify players about an accusation
+        else if (statUp instanceof AccuseNotification)
```
Been trying to offload these into separate functions like in the server because this function was getting huge

NBlair52 replied:
Yeah appreciate keeping it clean. Moved.

NBlair52 committed:
Kit Comments Addressed

Made processAccuseNotification
Made EnvelopePeakNotification with handling on both server and client 
side for when the person gets it wrong so they can see what it should 
have been.

Other minor changes

NBlair52 committed:
minor

Kit-Kindred commented on Source_Code/Client/ClueLessClient.java:
```diff
@@ -370,6 +403,25 @@ else if( statUp instanceof BoardUpdate)
+            processAccuseNotification((AccuseNotification) statUp);
+        }
+        //This is sent to the player when they accuse
+        else if (statUp instanceof EnvelopePeakNotification)
```
I had to comment this out and then replace the "instanceof Notification" block with 

`else if(statUp instanceof Notification)  // Generic server print essentially`
`        {`
`            if((Notification) statUp instanceof EnvelopePeakNotification)`
 `           {`
 `               System.out.println((EnvelopePeakNotification) statUp);`
 `           }`
 `           else`
 `           {`
 `               System.out.println("[Server] " + ((Notification) statUp).NotificationText);`
`            }`
`        }`

NBlair52 committed:
Fixed Notification to StatusUpdate

Put 9 option in ocrrect spot

NBlair52 commented:
Ready for real this time
